### PR TITLE
created BookController and BookDto classes. Extended BookRepository w…

### PR DESCRIPTION
**Pull request by vaniii1**
…ith new method

### CODE REVIEW

File: src/main/java/com/example/onlinebookstore/exception/EntityNotFoundException.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.example.onlinebookstore.exception;
+
+public class EntityNotFoundException extends RuntimeException {
+    public EntityNotFoundException(String message, Throwable cause) {
+        super(message, cause);
+    }
+
+    public EntityNotFoundException() {
+        super();
+    }
```
Comment (aaananas) [inline src/main/java/com/example/onlinebookstore/exception/EntityNotFoundException.java]:
```suggestion
    public EntityNotFoundException(String message, Throwable cause) {
        super(message, cause);
    }
```

File: src/main/java/com/example/onlinebookstore/mapper/BookMapperImpl.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
+package com.example.onlinebookstore.mapper;
+
+import com.example.onlinebookstore.dto.BookDto;
+import com.example.onlinebookstore.dto.CreateBookRequestDto;
+import com.example.onlinebookstore.model.Book;
+import org.springframework.stereotype.Component;
+
+@Component
+public class BookMapperImpl implements BookMapper {
```
Comment (aaananas) [inline src/main/java/com/example/onlinebookstore/mapper/BookMapperImpl.java]: this class should be generated in target folder

File: src/main/java/com/example/onlinebookstore/service/impl/BookServiceImpl.java [hunk 3]
```diff
@@ -1,24 +1,39 @@
...
+        return bookRepository.findAll()
+                .stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public BookDto getBookById(Long id) {
+        Book book = bookRepository.findBookById(id).orElseThrow(EntityNotFoundException::new);
+        return bookMapper.toDto(book);
```
Comment (aaananas) [inline src/main/java/com/example/onlinebookstore/service/impl/BookServiceImpl.java]:
```suggestion
        return bookRepository.findBookById(id)
                            .map(bookMapper::toDto)
                            .orElseThrow(() -> new EntityNotFoundException("your message");
```
just a suggestion

**Review by aaananas — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/onlinebookstore/mapper/BookMapperImpl.java (hunk 2 above)
Comment (vaniii1) [reply src/main/java/com/example/onlinebookstore/mapper/BookMapperImpl.java]: then checkstyle test will fail

Comment (kozhukhovsky) [reply src/main/java/com/example/onlinebookstore/mapper/BookMapperImpl.java]: It is necessary to exclude the check of the target directory from the checkstyle.
Write to slack, they will help you how to do it.

**Review by kozhukhovsky — CHANGES_REQUESTED** [review]
Need remove BookMapperImpl

**Review by olekskov — APPROVED** [review]

**Pull request merged**